### PR TITLE
[v2.9] Revert changes to hosted providers gke and eks clusterconfigs

### PR DESCRIPTION
**Pull request by cpinjani**
Cherry-pick of PR https://github.com/rancher/shepherd/pull/289 for release/v2.9 branch.

### CODE REVIEW

File: extensions/clusters/gke/gke_cluster_config.go [hunk 1]
```diff
@@ -143,10 +143,14 @@ func clusterAddonsBuilder(clusterAddons *ClusterAddons) *management.GKEClusterAd
 }
 
 func cmekConfigBuilder(cmekConfig *CMEKConfig) *management.CMEKConfig {
-	return &management.CMEKConfig{
-		KeyName:  cmekConfig.KeyName,
-		RingName: cmekConfig.RingName,
+	var CMEKConfig *management.CMEKConfig
+	if cmekConfig != nil {
```
Comment (igomez06) [inline extensions/clusters/gke/gke_cluster_config.go:147]: This needs the same fix Chandan as the main branch

**Review by igomez06 — CHANGES_REQUESTED** [review]
Do we need a v2.8 back port?

File: extensions/clusters/gke/gke_cluster_config.go (hunk 1 above)
Comment (cpinjani) [reply extensions/clusters/gke/gke_cluster_config.go]: Done, thanks.

**Comment by cpinjani** [comment]
> Do we need a v2.8 back port?

No Israel, [code](https://github.com/rancher/shepherd/blob/release/v2.8/pkg/schemas/management.cattle.io/v3/schema.go#L273) is already there.
Thanks for pointing it.

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by caliskanugur — APPROVED** [review]
LGTM

**Pull request merged**